src/popup-page/context/global.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { createContext, useReducer } from 'react'
+
+export class GlobalStateActionError extends Error {
+  constructor(action: string, message: string) {
+    const formattedMessage = `Error happened on action "${action}": ${message} `
+    super(formattedMessage)
+  }
+}
+
+interface IState {
+  beeApiUrl: string
+}
+
+interface IAction<T = string> {
+  type: 'BEE_API_URL_CHANGE'
+  newValue?: T
+}
+
+interface ContextValue {
+  state: IState
+  dispatch: React.Dispatch<IAction<string>>
+}
+
+const initialState: IState = {
+  beeApiUrl: 'http://localhost:1633',
+}
+const GlobalContext = createContext<ContextValue>({
+  state: initialState,
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  dispatch: () => {},
+})
+const { Provider } = GlobalContext
+
+const GlobalStateProvider = ({ children }: { children: React.ReactElement }): React.ReactElement => {
+  const [state, dispatch] = useReducer((state: IState, action: IAction): IState => {
+    switch (action.type) {
+      case 'BEE_API_URL_CHANGE':
+        if (!action.newValue || typeof action.newValue !== 'string') {
+          throw new GlobalStateActionError(action.type, 'No "newValue" property has been passed')
+        } else if (action.newValue.startsWith('http://') || action.newValue.startsWith('https://')) {
+          throw new GlobalStateActionError(action.type, '"newValue" is not start with either "http://" or "https://"')
+        }
```

Comment:
This also makes it case insensitive (which URLs are)
```suggestion
        } else if (/^https?:\/\//i.test(action.newValue) {
          throw new GlobalStateActionError(action.type, '"newValue" does not start with either "http://" or "https://"')
        }
```

Reply:
If we are dealing with regex it can check more
```suggestion
        }

        if (!/^https?:\/\/[a-zA-Z0-9\.]+/i.test(action.newValue)) {
          throw new GlobalStateActionError(action.type, '"newValue" is not start with either "http://" or "https://"')
        }
```

also had a problem with the negation